src/app.md

```diff
@@ -276,9 +276,7 @@ fn run() -> Result<()> {
 <a name="ex-file-skip-dot"></a>
 ##  Traverse directories while skipping dotfiles
 
-Iterates over filesystem entries efficiently skipping hidden files and directories with help of [WalkDir::filter_entry]. 
-
-[`WalkDir::filter_entry`] descends recursively into entries passing the `is_not_hidded` predicate. [`std::iter::Iterator::filter`] would be applied to each `DirEntry` even if the parent is a hidden directory. Uses [`WalkDir::depth`] to avoid skipping root directory `"."` at level 0.
+[`WalkDir::filter_entry`] descends recursively into entries passing the `is_not_hidded` predicate skipping hidden files and directories except the root directory `"."` at level 0 checked with [`WalkDir::depth`] whereas [`Iterator::filter`] would be applied to each `DirEntry` even if the parent is a hidden directory. 
```

Comment:
I would start the paragraph with "Uses [`WalkDir::filter_entry`] to descend" and move the "except the root directory `"."` at level 0 checked with [`WalkDir::depth`]" to its own sentence at the bottom.